src/test/java/org/assertj/core/api/InstanceOfAssertFactoriesTest.java

```diff
@@ -0,0 +1,105 @@
+package org.assertj.core.api;
+
+import static java.util.Arrays.asList;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.*;
+
+import java.security.Certificate;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.DoublePredicate;
+import java.util.function.IntPredicate;
+import java.util.function.LongPredicate;
+import java.util.function.Predicate;
+
+import org.assertj.core.util.Strings;
+import org.junit.jupiter.api.Test;
+
+public class InstanceOfAssertFactoriesTest {
+
+  @Test
+  public void predicate_factory_should_allow_predicate_assertions() {
+    Predicate<Object> predicate = Objects::isNull;
+    assertThat((Object) predicate).instanceOf(PREDICATE).accepts((Object) null);
+  }
+
+  @Test
+  public void typed_predicate_factory_should_allow_typed_predicate_assertions() {
+    Predicate<String> predicate = Strings::isNullOrEmpty;
+    assertThat((Object) predicate).instanceOf(predicate(String.class)).accepts("");
```

Comment:
I'm not completely happy on how the typed factories should be used. Specifically, the current implementation has a type parameter which is actually not used for the factory creation but just for type inference:
```java
public static <T> InstanceOfAssertFactory<Predicate, PredicateAssert<T>> predicate(Class<T> type) {
    return new InstanceOfAssertFactory<>(Predicate.class, PredicateAssert<T>::new);
}
```

If I drop the unused parameter, then the caller must specify the type parameter explicitly:

```java
// With unused parameter
assertThat((Object) predicate).instanceOf(predicate(String.class)).accepts("");
// Without parameter
assertThat((Object) predicate).instanceOf(InstanceOfAssertFactories.<String>predicate()).accepts("");
```

I tend to favor the unused parameter to have better readability.
@joel-costigliola what do you think?

Reply:
I agree, `assertThat((Object) predicate).instanceOf(predicate(String.class)).accepts("");` is way better than  `assertThat((Object) predicate).instanceOf(InstanceOfAssertFactories.<String>predicate()).accepts("");` which is not elegant at all.